packages/eas-cli/src/commands/workflow/view.ts

```diff
@@ -0,0 +1,127 @@
+import { getWorkflowRunUrl } from '../../build/utils/url';
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASNonInteractiveFlag, EasJsonOnlyFlag } from '../../commandUtils/flags';
+import { WorkflowRunByIdWithJobsQuery } from '../../graphql/generated';
+import { AppQuery } from '../../graphql/queries/AppQuery';
+import { WorkflowRunQuery } from '../../graphql/queries/WorkflowRunQuery';
+import Log, { link } from '../../log';
+import { promptAsync } from '../../prompts';
+import formatFields from '../../utils/formatFields';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
+
+export default class WorkflowView extends EasCommand {
+  static override description =
+    'view details for a workflow run, including jobs. If no run ID is provided, you will be prompted to select from recent workflow runs for the current project.';
+
+  static override flags = {
+    ...EasJsonOnlyFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override args = [{ name: 'id', description: 'ID of the workflow run to view' }];
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectId,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { args, flags } = await this.parse(WorkflowView);
+    const nonInteractive = flags['non-interactive'];
+    const {
+      loggedIn: { graphqlClient },
+      projectId,
+    } = await this.getContextAsync(WorkflowView, {
+      nonInteractive,
+    });
+    if (flags.json) {
+      enableJsonOutput();
+    }
+
+    let idToQuery = args.id;
+    if (!idToQuery) {
+      if (nonInteractive) {
+        throw new Error('If non-interactive, this command requires a workflow job ID as argument');
+      }
+      const runs = await AppQuery.byIdWorkflowRunsFilteredByStatusAsync(
```

Comment:
Would it make sense to make some utility for this? I see this repeated in `:logs` and `:view`.

Reply:
Done